### PR TITLE
ci: run test via make & remove travis CI config

### PR DESCRIPTION
**Pull request by fho**
```
ci: remove travis ci config

Travis is not used anymore, github actions is used instead

-------------------------------------------------------------------------------
ci: run tests via make

Run the tests in github actions via "make tests" instead of duplicating
the command to run the tests.

This makes it easier to run the tests the same way locally and keep the
commands in sync.

The makefile tests target is changed to pass "-cpu=1,2", to run the
tests in CI the same way then before.
```

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -16,4 +16,4 @@ vet: ## Run go vet against code
 
 .PHONY: tests
 tests: ## Run all tests and requires a running rabbitmq-server
-	env AMQP_URL=amqp://guest:guest@127.0.0.1:5672/ go test -race -v -tags integration
+	env AMQP_URL=amqp://guest:guest@127.0.0.1:5672/ go test -cpu 1,2 -race -v -tags integration
```
Comment (fho) [inline Makefile:19]: Why are the tests actually limited to run only on core 1 & 2?

Comment (lukebakken) [reply Makefile:19]: I have no idea! Maybe to reduce contention with RabbitMQ? It's probably not necessary.

Comment (Zerpet) [reply Makefile:19]: There's no additional context in the initial commit that set the `-cpu` flag 87488be91988d8e38f0d9f5c43f33bd5d2174e37. I agree it's probably not necessary.

Comment (lukebakken) [reply Makefile:19]: AHA! I bet this is why: https://splice.com/blog/lesser-known-features-go-test/

> By default, unless GOMAXPROCS is set via an environment variable or explicity in the code, the tests are run with GOMAXPROCS=1. One can type GOMAXPROCS=2 go test to test with 2 CPUs, and then type GOMAXPROCS=4 go test to test with 4, but there is a better way: go test -cpu=1,2,4 will run the tests three times, with 1, 2 and 4 as GOMAXPROCS values.

**Comment by fho** [comment]
Could somebody rerun the CI tests?
I think the failure is because the testcase is flaky, the same testcase is failing locally for me sometimes.

**Comment by lukebakken** [comment]
We'll have to get the flaky tests fixed before the next release.

**Comment by fho** [comment]
mmh, maybe `-race` changes the timing a bit and it causes the test failure

**Comment by lukebakken** [comment]
`TestRepeatedChannelExceptionWithPublishAndMaxProcsIssue46` is flaky in my local environment as well. If I can't figure out why soon I'll open a separate issue to fix it.

**Review by lukebakken — APPROVED** [review]

**Comment by lukebakken** [comment]
Thank you again!

**Pull request merged**